### PR TITLE
Remove TrustAnchorWithDn type

### PR DESCRIPTION
**Pull request by djc**
TBD: do we still need/want the `RootCertStore` type after this, or can we make do with a `Vec<TrustAnchor<'static>>`? I think we'll still need a place to wrap `webpki::extract_trust_anchor()`, and this eliminates the annoying allocations required to convert a `Vec<TrustAnchorWithDn<'_>>` to a `&[TrustAnchor<'_>]` so keeping it probably makes sense?

### CODE REVIEW

File: rustls/src/msgs/handshake.rs [hunk 1]
```diff
@@ -1687,6 +1687,22 @@ wrapped_payload!(
...
+impl DistinguishedName {
+    /// Create a [`DistinguishedName`] after prepending its outer SEQUENCE encoding.
+    ///
+    /// This can be decoded using [x509-parser's FromDer trait](https://docs.rs/x509-parser/latest/x509_parser/prelude/trait.FromDer.html).
+    ///
+    /// ```ignore
+    /// use x509_parser::prelude::FromDer;
+    /// println!("{}", x509_parser::x509::X509Name::from_der(dn.as_ref())?.1);
+    /// ```
+    pub fn in_sequence(bytes: &[u8]) -> Self {
```
[the remark this answers is not in the document]
Comment (ctz) [reply rustls/src/msgs/handshake.rs:1699]: I can think of lots of other names, but no convincing arguments why they'd be better than this :)

**Review by ctz — APPROVED** [review]
> TBD: do we still need/want the `RootCertStore` type after this, or can we make do with a `Vec<TrustAnchor<'static>>`?

I feel like the two remaining methods get quite a bit of use and are generally useful -- so I would like to see them kept. I guess they could become free functions that mutate a `Vec<TrustAnchor>`? The other option would be an extension trait on `Vec<TrustAnchor>`, which I generally find to be pretty annoying from an API discoverability point of view.

Given the item in `RootCertStore` is and always has been `pub`, people can move into/out of this type pretty freely to use these functions if desired?

File: rustls/src/msgs/handshake.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cpu) [reply rustls/src/msgs/handshake.rs:1699]: Seems like a perfectly cromulent name as-is :+1:

**Review by cpu — APPROVED** [review]
LGTM

**Pull request merged**